compiler/rustc_mir_transform/src/check_finalizers.rs

```diff
@@ -13,6 +13,82 @@ use rustc_trait_selection::infer::TyCtxtInferExt;
 #[derive(PartialEq)]
 pub struct CheckFinalizers;
 
+#[derive(Debug)]
+enum FinalizerErrorKind<'tcx> {
+    /// Does not implement `Send` + `Sync`
+    ThreadUnsafe(Span),
```

Comment:
Maybe `NotSendAndSync`?

Reply:
Sure. For consistency, should I replace `ContainsReference` with `NotReferenceFree`?

Reply:
Good idea.

Reply:
Fixed 9f10c609755fafec549f096d7d488481848786f1

Reply:
> Good idea.

Also fixed 622d07478b126de25c92fd048c350d4efcd30d6e